contrib/dimfeld/httptreemux.v5/httptreemux.go

```diff
@@ -115,17 +115,40 @@ func getRoute(router *httptreemux.TreeMux, w http.ResponseWriter, req *http.Requ
 
 	// Check for redirecting route due to trailing slash for parameters.
 	// The redirecting behaviour originates from httptreemux router.
-	if lr.StatusCode == http.StatusMovedPermanently && strings.HasSuffix(route, "/") {
-		rReq := req.Clone(req.Context())
-		rReq.RequestURI = strings.TrimSuffix(rReq.RequestURI, "/")
-		rReq.URL.Path = strings.TrimSuffix(rReq.URL.Path, "/")
-
-		lr, found = router.Lookup(w, rReq)
-		if !found {
-			return "", false
+	// TODO: what about the other two router.RedirectBehavior cases?
+	if lr.StatusCode == http.StatusMovedPermanently && router.RedirectBehavior == httptreemux.Redirect301 {
+		if strings.HasSuffix(route, "/") {
```

Comment:
This is for the original case where:
- request URL: `GET /api/example/`
- matched handler: `GET /api/example`